packages/ecommerce-app-base/src/types/product.ts

```diff
@@ -26,5 +26,6 @@ type ProductsFnResponse<P extends Product = Product> = {
 
 export type ProductsFn<P extends Product = Product> = (
   search: string,
-  pagination?: Partial<Pagination>
+  pagination?: Partial<Pagination>,
+  searchBySku?: boolean
```

Comment:
I think your concerns are warranted here @whitelisab but don't see a way around having these optional paramaters, so opted to keep as the way you had it!